src/main/scala/Compiler.scala

```diff
@@ -106,6 +107,24 @@ class EssentEmitter(initialOpt: OptFlags, writer: Writer) extends LazyLogging {
       }
       case _ => {
         writeLines(indentLevel, emitStmt(stmt))
+        if (opt.vcdOn) {
+          stmt match {
+          case mw: MemWrite =>
+          case _ => 
+            val resultName = findResultName(stmt)
+            resultName match {
+              case Some(name) =>
+              val cleanName = name.replace('.','$')
```

Comment:
Is there a way to tidy this up by using the Renamer? Perhaps small changes to Renamer could make this better?

Reply:
have used rn.removeDots function present in renamer